testV5/eth2DaiReserve.js

```diff
@@ -0,0 +1,1441 @@
+const Web3 = require('web3');
+const web3 = new Web3(new Web3.providers.HttpProvider('http://localhost:8545'));
+
+let TestToken = artifacts.require("TestTokenV5.sol");
+let WethToken = artifacts.require("WethTokenV5.sol");
+let MockOtcOrderbook = artifacts.require("MockOtcOrderbook.sol");
+let Eth2DaiReserve = artifacts.require("Eth2DaiReserve.sol");
+
+let Helper = require("../test/helper.js");
+let BigNumber = require('bignumber.js');
+
+const ethAddress = '0x00eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee';
+const precision = new BigNumber(10).pow(18);
+const feeBps = new BigNumber(25);
+
+const initOTCDaiBalance = new BigNumber(2000000).mul(BigNumber(10).pow(18));
+const initOTCWethBalance = new BigNumber(1000).mul(BigNumber(10).pow(18));
+
+const minDaiBal  = (new BigNumber(1000)).mul(BigNumber(10).pow(18))
+const maxDaiBal  = (new BigNumber(10000)).mul(BigNumber(10).pow(18))
+const initDaiBal = (new BigNumber(5000)).mul(BigNumber(10).pow(18))
+const minSpreadInBps = 70;
+const permiumBps     = 10;
+
+const maxTraverse = 20;
+const maxTraverseX = 4160;
+const maxTraverseY = 493146;
+const maxTake = 10;
+const maxTakeX = 2560;
+const maxTakeY = 260431;
+const minSupport = (new BigNumber(10)).pow(17); // 0.1 ether
+const minSupportX = 7000;
+const minSupportY = 193750;
+
+let admin;
+let alerter;
+let operator;
+let otc;
+let reserve;
+let network;
+let user;
+
+let myWethToken;
+let myDaiToken;
+
+contract('Eth2DaiReserve', function(accounts) {
+    before("one time init", async() => {
+        admin = accounts[0];
```

Comment:
why using same account for all functions?
it might be an issue for some tests. no?

Reply:
For now it doesn't have any issues for any tests

Reply:
Yes I have separated them, each for different accounts